internal/locate/replica_selector.go

```diff
@@ -300,6 +315,15 @@ func (s *ReplicaSelectMixedStrategy) next(selector *replicaSelector) *replica {
 	return nil
 }
 
+func (s *ReplicaSelectMixedStrategy) canSendReplicaRead(selector *replicaSelector) bool {
```

Comment:
Another approach is to integrate the following logic with 
```
if !s.target.store.IsLabelsMatch(s.option.labels) && s.target.peer.Id != s.region.GetLeaderPeerID()
```
,  and abstract a method like
```
func (s replicaSelector*) DecideReplicaReadType() ReadType {
    ...
}

if readType == staleRead {
    req.StaleRead = true
    req.ReplicaRead = false
} else if read Type == followerRead {
    ....
} else {
   ...
}
```